### PR TITLE
vc1d: fix frame type reporting

### PR DESCRIPTION
**Pull request by akokoshn**
-Add handling P/B frame types

### CODE REVIEW

**Comment by akokoshn** [comment]
WIP

File: _studio/mfx_lib/decode/vc1/src/mfx_vc1_decode.cpp [hunk 1]
```diff
@@ -49,6 +49,12 @@ static void SetFrameType(const UMC::VideoData &mediaData, mfxFrameSurface1 &surf
         case UMC::I_PICTURE:
             extFrameInfo->FrameType = MFX_FRAMETYPE_I;
             break;
+        case UMC::P_PICTURE:
+            extFrameInfo->FrameType = MFX_FRAMETYPE_P;
+            break;
+        case UMC::B_PICTURE:
```
Comment (lizhong1008) [inline _studio/mfx_lib/decode/vc1/src/mfx_vc1_decode.cpp]: VC1 has B and BI type.  Probably the issue you commented at https://github.com/Intel-Media-SDK/MediaSDK/issues/333#issuecomment-468352510 is caused BI type is not handled here.

Comment (akokoshn) [reply _studio/mfx_lib/decode/vc1/src/mfx_vc1_decode.cpp]: BI-Frames handled as B:
`case  VC1_B_FRAME:
 case  VC1_BI_FRAME:
     out_data->SetFrameType(B_PICTURE);
     break;`
umc_vc1_video_decoder_hw.h: line 169

Comment (lizhong1008) [reply _studio/mfx_lib/decode/vc1/src/mfx_vc1_decode.cpp]: Hm, I see. So what is the frame_tpye which is still reported as unknown?

File: _studio/shared/umc/codec/vc1_dec/include/umc_vc1_video_decoder_hw.h [hunk 2]
```diff
@@ -157,10 +157,22 @@ namespace UMC
...
+                case  VC1_P_FRAME:
+                    out_data->SetFrameType(P_PICTURE);
+                    break;
+                case  VC1_B_FRAME:
+                case  VC1_BI_FRAME:
+                    out_data->SetFrameType(B_PICTURE);
+                    break;
+                case  VC1_SKIPPED_FRAME:
+                    out_data->SetFrameType(D_PICTURE); // means skipped
+                    break;
```
Comment (dvrogozh) [inline _studio/shared/umc/codec/vc1_dec/include/umc_vc1_video_decoder_hw.h]: default is missing. Can we potentially come into it here?

Comment (akokoshn) [reply _studio/shared/umc/codec/vc1_dec/include/umc_vc1_video_decoder_hw.h]: Default value (NONE) sets on initialization  out_data, we can reset NONE here, but I guess it's extra action

File: _studio/mfx_lib/decode/vc1/src/mfx_vc1_decode.cpp (hunk 1 above)
Comment (akokoshn) [reply _studio/mfx_lib/decode/vc1/src/mfx_vc1_decode.cpp]: VC1_SKIPPED_FRAME reported as unknown, should be fixed in the latest update this PR

File: _studio/shared/umc/codec/vc1_dec/include/umc_vc1_video_decoder_hw.h [hunk 3]
```diff
@@ -157,10 +157,19 @@ namespace UMC
...
+                case  VC1_I_FRAME:
+                    out_data->SetFrameType(I_PICTURE);
+                    break;
+                case  VC1_P_FRAME:
+                    out_data->SetFrameType(P_PICTURE);
+                    break;
+                case  VC1_B_FRAME:
+                case  VC1_BI_FRAME:
+                    out_data->SetFrameType(B_PICTURE);
+                    break;
```
Comment (dvrogozh) [inline _studio/shared/umc/codec/vc1_dec/include/umc_vc1_video_decoder_hw.h]: You know, I am nervous when I see switch without the default case. This is origin of errors. Hence this is the wonderful place to add `assert()` even if you believe we won't be here.

**Comment by akokoshn** [comment]
@dvrogozh, sure, I will provide test results

**Pull request merged**